pkg/yurthub/restconfig/restconfig.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restconfig
+
+import (
+	"net/url"
+
+	"github.com/openyurtio/openyurt/cmd/yurthub/app/config"
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+	"github.com/openyurtio/openyurt/pkg/yurthub/certificate/interfaces"
+	"github.com/openyurtio/openyurt/pkg/yurthub/healthchecker"
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+type RestConfigManager struct {
+	remoteServers []*url.URL
+	certMgrMode   string
+	checker       healthchecker.HealthChecker
+	certManager   interfaces.YurtCertificateManager
+}
+
+// NewRestConfigManager creates a *RestConfigManager object
+func NewRestConfigManager(cfg *config.YurtHubConfiguration, certMgr interfaces.YurtCertificateManager, healthChecker healthchecker.HealthChecker) (*RestConfigManager, error) {
```

Comment:
i think we need to add some unit test cases for `RestConfigManager`?

Reply:
The latest code has been commited. Unit tests has been added.